cfme/tests/infrastructure/test_vmware_provider.py

```diff
@@ -249,7 +254,31 @@ def test_vmware_cdrom_dropdown_not_blank():
             4.Virtual machine is selected
             5.Dropdown of ISO files is not empty for CD/DVD Drive
     """
-    pass
+    datastore_collection = appliance.collections.datastores
+    ds = [ds.name for ds in provider.data['datastores'] if ds.type == 'iso']
+    try:
+        iso_ds = datastore_collection.instantiate(name=ds[0], provider=provider)
+    except IndexError:
+        pytest.skip('No datastores found of type iso on provider {}'.format(provider.name))
+    iso_ds.run_smartstate_analysis()
+    vms_collections = appliance.collections.infra_vms
+    vm = vms_collections.instantiate(name='cu-24x7', provider=provider)
+    view = navigate_to(vm, 'Reconfigure')
+    actions_column = view.cd_dvd_table[0]['Actions']
+    # First disconnect if already connected
+    if actions_column.text == 'Connect Disconnect':
+        actions_column.click()
+    # Confirm disconnect
+    if actions_column.text == 'Confirm':
+        actions_column.click()
+    # if 'Connect' option is present, click it
+    if actions_column.text == 'Connect':
+        actions_column.click()
```

Comment:
Is there a chance it isn't any of these? It would be good to assert that it is connected.

Reply:
There will be something. It is usually Connected, hence trying to disconnect first. Certainly there is implicit assumption there too. But for what I saw, it was always connected. So flow I assumed is `Disconnect`->`Confirm`->`(Re)Connect`
Do we really need to assert it? wouldn't the `if` conditions take care of it?

Reply:
Adding a try/except for actions_column, for some VMs the table is empty I noticed.

Reply:
This was discussed offline. @john-dupuy and I did not see a better way for this. Maybe @mshriver or @izapolsk has a better one for these multiple if statements

Reply:
Yeah this is certainly an odd pattern.  If these `if` statements aren't `True`, then the test isn't actually clicking the button, and there's no indication in the test output.

I think it would be better to follow a pattern that actually asserts the expected states, not just an `if` check.

Reading this, it seems you could collapse it to `if actions_column.text in ['Connect', 'Confirm', 'Connect Disconnect']: actions_column.click()`.  This isn't the case though, because you're actually expecting this test to click the action column multiple times.

I think it should be more like:
```
assert actions_column.text == 'Connect Disconnect', 'CD/DVD action button not in expected state'
actions_column.click()
assert actions_column.text == 'Confirm', 'CD/DCD action button did not change to "Confirm"'
# etc
```